definitions/npm/react-apollo_v2.x.x/flow_v0.58.x-/react-apollo_v2.x.x.js

```diff
@@ -276,16 +276,16 @@ declare module 'react-apollo' {
 
   declare export class Subscription<TData> extends React$Component<SubscriptionProps<TData>> {}
 
-  declare export type MutationFunction<TVariables> = (options: {
+  declare export type MutationFunction<TData=void, TVariables=void> = (options: {
     variables?: TVariables,
     optimisticResponse?: Object,
     refetchQueries?: (mutationResult: FetchResult) => Array<{query: DocumentNode, variables: {[string]: any}}>,
     update?: (cache: DataProxy, mutationResult: FetchResult) => any
-  }) => Promise<*>
+  }) => Promise<MutationResult<TData>>
 
-  declare export type MutationResult<TData> = {loading: boolean, error?: ApolloError, data?: TData, called: boolean}
+  declare export type MutationResult<TData=void> = {loading: boolean, error?: ApolloError, data?: TData, called: boolean}
```

Comment:
Think this void should similary be `{ [key: string]: any }` (https://github.com/apollographql/react-apollo/blob/master/src/Mutation.tsx#L12)